### PR TITLE
refactor: rename to LSP23LinkedContractsFactory

### PR DESCRIPTION
**Pull request by skimaharvey**
# What does this PR introduce? 

Rename `LinkedContractsFactory` to `LSP23LinkedContractsFactory`

### CODE REVIEW

**Review by b00ste — APPROVED** [review]

File: tests/LSP23LinkedContractsDeployment/LSP23LinkedContractsDeployment.test.ts [hunk 1]
```diff
@@ -202,15 +204,15 @@ describe('UniversalProfileDeployer', function () {
         );
 
       /* deploying gives a gas error while static calls work fine */
-      // await LinkedContractsFactory.deployContracts(
+      // await LSP23LinkedContractsFactory.deployContracts(
```
Comment (CJ42) [inline tests/LSP23LinkedContractsDeployment/LSP23LinkedContractsDeployment.test.ts:207]: is this intended to be commented?

**Review by CJ42 — CHANGES_REQUESTED** [review]
Please do not forget to also change the name in `hardhat.config.ts`. Otherwise the JSON artifacts will not be present in the npm package!

https://github.com/lukso-network/lsp-smart-contracts/blob/614cec6c07cdef74b493f2d04a61374acacd1e8d/hardhat.config.ts#L153

File: tests/LSP23LinkedContractsDeployment/LSP23LinkedContractsDeployment.test.ts (hunk 1 above)
Comment (skimaharvey) [reply tests/LSP23LinkedContractsDeployment/LSP23LinkedContractsDeployment.test.ts:207]: Yes I can remove it but basically it is not working when deploying two contracts at once. I am guessing it is something wrong with hardhat node. 
the `callstatic` does work though

Comment (skimaharvey) [reply tests/LSP23LinkedContractsDeployment/LSP23LinkedContractsDeployment.test.ts:207]: wanted to leave the comment to show that I tried. Also if you guys can try on your end that would be awesome ;)

**Review by YamenMerhi — APPROVED** [review]
Need to change also in `hardhat.config.ts`

File: tests/LSP23LinkedContractsDeployment/LSP23LinkedContractsDeployment.test.ts (hunk 1 above)
Comment (CJ42) [reply tests/LSP23LinkedContractsDeployment/LSP23LinkedContractsDeployment.test.ts:226]: resolved in last commit ✅

**Review by CJ42 — APPROVED** [review]

**Review by CJ42 — APPROVED** [review]

**Pull request merged**